docs/source/checks/nlp/data_integrity/plot_frequent_substrings.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+"""
+.. _nlp__frequent_substrings:
+
+Frequent Substrings
+********************
+
+This notebook provides an overview for using and understanding the frequent substrings check:
+
+**Structure:**
+
+* `Why check for frequent substrings? <#why-check-for-text-data-duplicates>`__
+* `Create TextData <#create-textdata>`__
+* `Run the Check <#run-the-check>`__
+* `Define a Condition <#define-a-condition>`__
+
+Why check for frequent substrings?
+===================================
+
+The ``FrequentSubstrings`` check finds frequent substrings of varying lengths (n-grams) in the Dataset.
+Frequent substrings may be an indicator for a problem in the data pipeline that requires attention.
```

Comment:
Would import to here most (or all) of the explanation of the algorithm that you added in the check docstring.
Also, please be a bit more specific about what "problem", and give an example of what you may want to do about it.
Would put the algorithm description after the paragraph talking about the "why"